src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -0,0 +1,325 @@
+  ) : void {
+    // Noop
+  },
+
+  shouldSetTextContent(props : Props) : boolean {
+    return (
+      typeof props.children === 'string' ||
+      typeof props.children === 'number'
+    );
+  },
+
+  resetTextContent(testElement : Instance) : void {
+    // Noop
+  },
+
+  createTextInstance(
+    text : string,
+    rootContainerInstance : Container,
+    hostContext : Object,
+    internalInstanceHandle : Object
+  ) : TextInstance {
+    var inst = {
+      text : text,
+      id: instanceCounter++,
+      rootContainerInstance,
+      toJSON: () => isNaN(+inst.text) ? inst.text : +inst.text,
+    };
+    // Hide from unit tests
+    Object.defineProperty(inst, 'id', { value: inst.id, enumerable: false });
+    return inst;
+  },
+
+  commitTextUpdate(textInstance : TextInstance, oldText : string, newText : string) : void {
+    textInstance.text = newText;
+  },
+
+  appendChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
+    parentInstance.appendChild(child);
+  },
+
+  insertBefore(
+    parentInstance : Instance | Container,
+    child : Instance | TextInstance,
+    beforeChild : Instance | TextInstance
+  ) : void {
+    parentInstance.insertBefore(beforeChild, child);
+  },
+
+  removeChild(parentInstance : Instance | Container, child : Instance | TextInstance) : void {
+    parentInstance.removeChild(child);
+  },
+
+  scheduleAnimationCallback(fn : Function) : void {
+    setTimeout(fn);
+  },
+
+  scheduleDeferredCallback(fn: Function) : void {
+    setTimeout(fn, 0, {timeRemaining: Infinity});
+  },
+
+  useSyncScheduling: true,
+
+  getPublicInstance(ref) {
+    const createNodeMock = ref.rootContainerInstance.createNodeMock;
+    return createNodeMock(ref);
+  },
+
+});
+
+var defaultTestOptions = {
+  createNodeMock: function() {
+    return null;
+  },
+};
+
+var ReactTestFiberRenderer = {
+  create(element: ReactElement<any>, options: TestRendererOptions) {
+    var createNodeMock = defaultTestOptions.createNodeMock;
+    if (options && typeof options.createNodeMock === 'function') {
+      createNodeMock = options.createNodeMock;
+    }
+    var container = new TestContainer(createNodeMock);
+    var root = TestRenderer.createContainer(container);
+    if (root) {
+      TestRenderer.updateContainer(element, root, null, null);
+    }
+    return {
+      toJSON() {
+        if (root == null) {
+          return null;
+        }
+        const hostInstance = TestRenderer.findHostInstance(root);
+        if (hostInstance === null) {
+          return hostInstance;
+        }
+        return hostInstance.toJSON();
+      },
+
+      update(newElement: ReactElement<any>) {
+        if (root == null) {
+          return;
+        }
+        TestRenderer.updateContainer(newElement, root, null, null);
+      },
+      unmount() {
+        if (root == null) {
+          return;
+        }
+        TestRenderer.updateContainer(null, root, null, () => {
+          container = null;
```

Comment:
I think you shouldn't need this callback.
Since it uses synch scheduling, everything should happen synchronously anyway.
So you should be able to clean up immediately after calling `updateContainer`.